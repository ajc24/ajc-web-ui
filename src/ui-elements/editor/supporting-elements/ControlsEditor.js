import React from 'react';
import PropTypes from 'prop-types';
import HorizontalButtonList from '../../form/HorizontalButtonList';

/**
 * The ControlsEditor component renders the controls / buttons for the editor panel of the document editor.
 */
class ControlsEditor extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    /* Set the buttons list for the editor controls */
    const upperEditorButtonsList = [
      {
        id: 'add-new-item-button',
        onClick: this.props.handleClickAddNewItem,
        title: 'Add New Item',
        tooltip: 'Add a new item to your document.',
        type: 'button',
      },
    ];
    const middleEditorButtonsList = [
      {
        id: 'save-as-draft-button',
        // onClick: this.props.onClickSaveAsDraft,
        title: 'Save as Draft',
        tooltip: 'Save your document in its current incomplete state. You may return to finish it later.',
        type: 'button',
      },
      {
        id: 'generate-preview-button',
        // onClick: this.props.onClickGeneratePreview,
        title: 'Generate Preview',
        tooltip: 'Generate a preview of your document in its current state.',
        type: 'button',
      },
      {
        id: 'publish-document-button',
        // onClick: this.props.onClickPublishDocument,
        title: 'Publish Document',
        tooltip: 'When you are finished with your document, click here to publish it.',
        type: 'button',
      },
    ];
    const lowerEditorButtonsList = [
      {
        id: 'discard-draft-button',
        // onClick: this.props.onClickDiscardDraft,
        title: 'Discard Draft',
        tooltip: 'Discard your current working document. Please note you will not be able to return to it after discarding it.',
        type: 'button',
      }
    ];
    return (
      <React.Fragment>
        <HorizontalButtonList alignment="left" additionalUpperSpacing={this.props.upperButtonListAdditionalSpacing} colour={this.props.colour}
          buttonList={upperEditorButtonsList} labelText="Click the button below to add new items to your document:" />
        <HorizontalButtonList alignment="left" additionalUpperSpacing={true} colour={this.props.colour} buttonList={middleEditorButtonsList}
          labelText="Save your current draft, generate a preview or publish your document:" />
        <HorizontalButtonList alignment="left" additionalUpperSpacing={true} colour={this.props.colour} buttonList={lowerEditorButtonsList}
          labelText="Click the button below to discard this current draft:" />
      </React.Fragment>
    );
  }
}
ControlsEditor.propTypes = {
  /** The colour / theme set to the document editor buttons and all relevant items in the preview page. */
  colour: PropTypes.oneOf([ 'grey', 'red' ]),
  /** The click functionality for the add new item button. */
  handleClickAddNewItem: PropTypes.func,
  /** The click functionality for the discard draft button. */
  // onClickDiscardDraft: PropTypes.func,
  // /** The click functionality for the generate preview button. */
  // onClickGeneratePreview: PropTypes.func,
  // /** The click functionality for the publish document button. */
  // onClickPublishDocument: PropTypes.func,
  // /** The click functionality for the save as draft button. */
  // onClickSaveAsDraft: PropTypes.func,
  /** Use additional spacing above the upper editor button list. */
  upperButtonListAdditionalSpacing: PropTypes.bool,
};
ControlsEditor.defaultProps = {
  colour: 'grey',
  handleClickAddNewItem: () => {},
  // onClickDiscardDraft: () => {},
  // onClickGeneratePreview: () => {},
  // onClickPublishDocument: () => {},
  // onClickSaveAsDraft: () => {},
  upperButtonListAdditionalSpacing: false,
};
export default ControlsEditor;
