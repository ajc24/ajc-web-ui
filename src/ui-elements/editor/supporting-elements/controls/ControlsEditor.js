import React from 'react';
import PropTypes from 'prop-types';
import ControlsEditorUpper from './supporting-elements/ControlsEditorUpper';

/**
 * The ControlsEditor component renders the controls / buttons for the editor panel of the document editor.
 */
class ControlsEditor extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    /* Set the buttons list for the editor controls */
    // const middleEditorButtonsList = [
    //   {
    //     id: 'save-as-draft-button',
    //     onClick: () => { alert('Clicked save as draft button.'); },
    //     title: 'Save as Draft',
    //     tooltip: 'Save your document in its current incomplete state. You may return to finish it later.',
    //     type: 'button',
    //   },
    //   {
    //     id: 'generate-preview-button',
    //     onClick: () => { alert('Clicked generate preview button.'); },
    //     title: 'Generate Preview',
    //     tooltip: 'Generate a preview of your document in its current state.',
    //     type: 'button',
    //   },
    //   {
    //     id: 'publish-document-button',
    //     onClick: () => { alert('Clicked publish document button.'); },
    //     title: 'Publish Document',
    //     tooltip: 'When you are finished with your document, click here to publish it.',
    //     type: 'button',
    //   },
    // ];
    // const lowerEditorButtonsList = [
    //   {
    //     id: 'discard-draft-button',
    //     onClick: () => { alert('Clicked discard draft button.'); },
    //     title: 'Discard Draft',
    //     tooltip: 'Discard your current working document. Please note you will not be able to return to it after discarding it.',
    //     type: 'button',
    //   }
    // ];
    return (
      <React.Fragment>
        <ControlsEditorUpper additionalSpacingAboveControls={this.props.additionalSpacingAboveControls} colour={this.props.colour}
          addEditorItem={this.props.addEditorItem} dialogContentAreaColour={this.props.dialogContentAreaColour} reRenderAllowance={this.props.reRenderAllowance} />
        {/* <HorizontalButtonList alignment="left" additionalUpperSpacing={true} colour={this.props.colour} buttonList={middleEditorButtonsList}
          labelText="Save your current draft, generate a preview or publish your document:" />
        <HorizontalButtonList alignment="left" additionalUpperSpacing={true} colour={this.props.colour} buttonList={lowerEditorButtonsList}
          labelText="Click the button below to discard this current draft:" /> */}
      </React.Fragment>
    );
  }
}
ControlsEditor.propTypes = {
  /** The functionality to be carried out to add a new item to the editor. */
  addEditorItem: PropTypes.func.isRequired,
  /** The colour / theme set to the document editor buttons and all relevant items in the preview page. */
  colour: PropTypes.oneOf([ 'grey', 'red' ]),
  /** The background colour for the content area of any dialogs rendered. By default this is set to white. */
  dialogContentAreaColour: PropTypes.oneOf([ 'grey', 'white', 'yellow' ]),
  /** The time allowance between a state change and a re-render occurring. */
  reRenderAllowance: PropTypes.number,
  /** Use additional spacing above the upper buttons. */
  additionalSpacingAboveControls: PropTypes.bool,
};
ControlsEditor.defaultProps = {
  colour: 'grey',
  dialogContentAreaColour: 'white',
  reRenderAllowance: 100,
  additionalSpacingAboveControls: false,
};
export default ControlsEditor;
