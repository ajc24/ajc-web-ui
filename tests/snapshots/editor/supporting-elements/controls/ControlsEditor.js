import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import ControlsEditor from '../../../../../src/ui-elements/editor/supporting-elements/controls/ControlsEditor';

describe('ControlsEditor', () => {
  describe('Default props and rendering', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <ControlsEditor addEditorItem={jest.fn()} />
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - All properties custom set', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <ControlsEditor colour="red" addEditorItem={jest.fn()} additionalSpacingAboveControls={true} dialogContentAreaColour="grey" reRenderAllowance={500} />
        </React.Fragment>
      );
    });
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
