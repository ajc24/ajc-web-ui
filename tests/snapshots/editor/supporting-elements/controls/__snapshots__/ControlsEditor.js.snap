// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlsEditor Default props and rendering verifies the snapshot for the component 1`] = `
Array [
  <div
    className="ajc-flex-row ajc-spacing-bottom-small ajc-width-hd-inner ajc-text-black ajc-font-default ajc-flex-justify-left"
  >
    <h2
      className="ajc-font-paragraph-heading"
    >
      Click the button below to add new items to your document:
    </h2>
  </div>,
  <div
    className="ajc-flex-row ajc-spacing-bottom-small ajc-background-transparent ajc-width-hd-inner ajc-flex-justify-left ajc-spacing-top-reduced"
  >
    <button
      className="ajc-button ajc-font-default ajc-button-grey"
      id="add-new-item-button"
      onClick={[Function]}
      title="Add a new item to your document."
      type="button"
    >
      Add New Item
    </button>
  </div>,
  <input
    className="ajc-display-none"
    id="add-new-item-user-value"
    readOnly={true}
    type="text"
  />,
]
`;

exports[`ControlsEditor Transferred props and rendering - All properties custom set verifies the snapshot for the component 1`] = `
Array [
  <div
    className="ajc-flex-row ajc-spacing-bottom-small ajc-width-hd-inner ajc-text-black ajc-font-default ajc-flex-justify-left ajc-spacing-top-large"
  >
    <h2
      className="ajc-font-paragraph-heading"
    >
      Click the button below to add new items to your document:
    </h2>
  </div>,
  <div
    className="ajc-flex-row ajc-spacing-bottom-small ajc-background-transparent ajc-width-hd-inner ajc-flex-justify-left ajc-spacing-top-reduced"
  >
    <button
      className="ajc-button ajc-font-default ajc-button-red"
      id="add-new-item-button"
      onClick={[Function]}
      title="Add a new item to your document."
      type="button"
    >
      Add New Item
    </button>
  </div>,
  <input
    className="ajc-display-none"
    id="add-new-item-user-value"
    readOnly={true}
    type="text"
  />,
]
`;
